sycl/doc/SYCL2020-SpecializationConstants.md

```diff
@@ -0,0 +1,874 @@
+# Specialization constants
+
+Specialization constants are implemented in accordance with how they are defined
+by SYCL 2020 specification: [SYCL registry][sycl-registry],
+[direct link to the specification][sycl-2020-spec].
+
+[sycl-registry]: https://www.khronos.org/registry/SYCL/
+[sycl-2020-spec]: https://www.khronos.org/registry/SYCL/specs/sycl-2020/html/sycl-2020.html
+
+TODO: feature overview? code example?
```

Comment:
I was pressed to 'LGTM' for a bit here... can we fix the "TODOs" before we do so?

Reply:
This particular `TODO` fixed in 0e756c1c3147b2a0b5d701037a2b868716c63346, but there are still at least one more left